src/components/Input/index.jsx

```diff
@@ -0,0 +1,107 @@
+import { useEffect, useRef, useState } from 'react';
+
+import validate from './validate';
+
+export default function Input({ type = null, state = '', ...props }) {
+  // For controlling input
+  const [inputValue, setInputValue] = useState('');
+
+  // Input settings
+  const [inputStatus, setInputStatus] = useState(''); // Success, error, ''
+  const [inputDesc, setInputDesc] = useState(''); // Description under input
+  const [label, setLabel] = useState(''); // Lable for input
+  const [iconPath, setIconPath] = useState(''); // Icon substitution
+  const inputRef = useRef(null); // To change the input type
+  const inputId = `${type}-${document.querySelectorAll('#subtags input[type=text]').length}`;
+
+  // To show/hide the password is in the input
+  const [isPassShow, setIsPassShow] = useState(false);
```

Comment:
Little bit of nitpicking.
Let's rename it to `passShow`, because `isPassShow` sounds like a function because it starts with a verb

Reply:
Hm, I've seen in tutorials that if we want to create a boolean variable that gives us a some state, it's better to use "is"